task8/ServletsOnMaven/src/main/java/ServletsOnPosts/OutputPostsServlet.java

```diff
@@ -1,36 +1,74 @@
 package ServletsOnPosts;
 
-import Posts.Post;
-import Posts.PostsContainer;
-import org.omg.CORBA.INTERNAL;
+import Posts.*;
+import Posts.UtilClasses.*;
+import WorkWithDB.WorkWithDB;
 
+import javax.annotation.Resource;
 import javax.servlet.ServletException;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import javax.sql.DataSource;
 import java.io.IOException;
 import java.io.PrintWriter;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 
 public class OutputPostsServlet extends HttpServlet {
+
+    @Resource(name="jdbc/mydb")
+    private DataSource dataSource;
+
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
-        String topParam = req.getParameter("top");
         String skipParam = req.getParameter("skip");
-        int top;
+        String countParam = req.getParameter("count");
         int skip;
-        if (topParam == null) {
-            top = 10;
-        } else {
-            top = Integer.parseInt(topParam);
-        }
+        int count;
         if (skipParam == null) {
             skip = 0;
         } else {
             skip = Integer.parseInt(skipParam);
         }
-        List<Post> posts = PostsContainer.getPosts(null, skip, top);
+        if (countParam == null) {
+            count = 10;
+        } else {
+            count = Integer.parseInt(countParam);
+        }
+        String authorParam = req.getParameter("author");
+        String leftDateParam = req.getParameter("left");
+        String rightDateParam = req.getParameter("right");
+        String hashTagsParam = req.getParameter("hashTags");
+        List<String> hashTags = hashTagsParam == null ? new ArrayList<>()
+                : new ArrayList<>(Arrays.asList(hashTagsParam.split("[\\s\\n\\t\\r]")));
+        hashTags.removeIf(item->item.equals(""));
+        hashTags.replaceAll(item->'#' + item);
+
+        long leftMilliSeconds = leftDateParam == null ? 0 :
+                Long.parseLong(leftDateParam);
+        long rightMilliSeconds = rightDateParam == null
+                ? LocalDateTime.now().toEpochSecond(ZoneOffset.UTC) * 1000
+                : Long.parseLong(rightDateParam);
+
+        FilterParams filterParams = new FilterParams(
```

Comment:
You ought to find way to reuse this code snippet from `GetSizeServlet` of vice versa, because code copypasting is never welcome